sessions/2018/december/Session_20.md

```diff
@@ -0,0 +1,955 @@
+# SOFTWARE ENGINEERING IN JAVA
+
+## Session 20 (18/12/2018)
+
+- Project creation for API application
+- Rebase / merge explanation
+- Reseting commits
+- See the amount of branches and the differences between the master
+- Resolving merge conflicts in API Application
+
+### Notes
+
+#### Project creation
+
+We created a new [project for the API application][1] in order to track the tasks using [kanban method][9].
+
+#### Calculator Frontend
+
+Roger was explaining to me how to use `rebase` and the difeference with `merge` for the [calculator-frontend repository][2].
+
+He was also explaining to me why he used an empty branch called `Initial codebase` with an initial commit called `delete-me-after-review` as the beggining of the history. He did that in order to he could raise this [Pull Request (PR) from master against that branch][3] and to compare seeing all changes I did in master, being able to do comments, but with the caveat to couldn't make reviews because was himself who created that branch and raised the PR as he states:
```

Comment:
The other way around

```suggestion
He was also explaining to me why he used an initial commit called `Initial codebase` on a branch called `delete-me-after-review` as the begining of the history. He did that in order to he could raise this [Pull Request (PR) from master against that branch][3] and to compare seeing all changes I did in master, being able to do comments, but with the caveat to couldn't make reviews because was himself who created that branch and raised the PR as he states:
```